#!/usr/bin/env python
PACKAGE = "ros_detection_legs"

from dynamic_reconfigure.parameter_generator_catkin import *
from ros_detection_legs.deep_learning.config import loader

default_values = loader.load_parameters()["prepocessing"]

gen = ParameterGenerator()

gen.add("limit_distance",      double_t, 0, "limit valid distance between 2 points", default_values["limit_distance"],      0,5)
gen.add("limit_cluster_valid", double_t, 0, "limit cluster is valid",                default_values["limit_cluster_valid"], 0,1)
gen.add("limit_jump",          int_t,    0, "number points can't be jumped",         default_values["limit_jump"],          0,200)
gen.add("limit_radius",        double_t, 0, "limit distance (only radius)",          default_values["limit_radius"],        0,5)
gen.add("limit_length_data",   int_t,    0, "limit data length",                     default_values["limit_length_data"],   0,200)

exit(gen.generate(PACKAGE, "ros_detection_legs", "HyperParameters"))